src/devtools/useAtomsDevtools.ts

```diff
@@ -0,0 +1,174 @@
+    }
+    result[atomToDebuggingString(atom)] = [...atomState.d.keys()].map(
+      atomToDebuggingString
+    )
+  })
+
+  return result
+}
+
+export function useAtomsDevtools(name: string, scope?: Scope) {
+  const snapshot = useAtomsSnapshot()
+  const goToSnapshot = useGotoAtomsSnapshot()
+  let extension: Extension | undefined
+  try {
+    extension = (window as any).__REDUX_DEVTOOLS_EXTENSION__ as Extension
+  } catch {}
+  if (!extension) {
+    if (
+      typeof process === 'object' &&
+      process.env.NODE_ENV === 'development' &&
+      typeof window !== 'undefined'
+    ) {
+      console.warn('Please install/enable Redux devtools extension')
+    }
+  }
+
+  const ScopeContext = getScopeContext(scope)
+  const scopeContainer = useContext(ScopeContext)
+  const store = scopeContainer.s
+
+  if (!store[DEV_SUBSCRIBE_STATE]) {
+    throw new Error('useAtomsSnapshot can only be used in dev mode.')
+  }
+
+  const isTimeTraveling = useRef(false)
+  const isRecording = useRef(true)
+  const devtools = useRef<ConnectionResult & { shouldInit?: boolean }>()
+  const snapshots = useRef<AtomsSnapshot[]>([])
+
+  useEffect(() => {
+    let devtoolsUnsubscribe: () => void | undefined
+    if (extension) {
+      devtools.current = extension.connect({ name })
+
+      devtoolsUnsubscribe = devtools.current.subscribe((message: Message) => {
+        switch (message.type) {
+          case 'ACTION':
+            return
+          case 'DISPATCH':
+            switch (message.payload?.type) {
+              case 'RESET':
+                // Todo
+                return
+              case 'COMMIT':
+                devtools.current!.init(undefined)
+                return
+              case 'JUMP_TO_STATE':
+              case 'JUMP_TO_ACTION':
+                isTimeTraveling.current = true
+
+                const snapshot =
+                  snapshots.current[message.payload.actionId - 1]!
+
+                goToSnapshot(snapshot)
+                return
+              case 'PAUSE_RECORDING':
+                return (isRecording.current = !isRecording.current)
+            }
+        }
+      })
+
+      devtools.current.shouldInit = true
+    }
+    return () => {
+      devtoolsUnsubscribe?.()
+    }
+  }, [store])
+
+  useEffect(() => {
+    if (!devtools.current) {
+      return
+    }
+    if (devtools.current.shouldInit) {
+      devtools.current.init(undefined)
+      devtools.current.shouldInit = false
+      return
+    }
+
+    if ([...snapshot.keys()].length === 0) {
+      return
+    }
+
+    if (isTimeTraveling.current) {
+      // better solution to stop logging on JUMP_TO_ACTION & JUMP_TO_STATE ?
+      setTimeout(() => {
+        isTimeTraveling.current = false
+      }, 0)
+    } else if (isRecording.current) {
+      snapshots.current.push(snapshot)
+
+      const parsedSnapshot = serializeSnapshot(snapshot)
+
+      devtools.current.send(
+        `action:${snapshots.current.length} - ${new Date().toLocaleString()}`,
```

Comment:
Should this be just
```
`${snapshots.current.length}`
```
to make L124 work?

If we want the date, put it in the action object.
```js
{
  type: `${snapshots.current.length}`,
  updatedAt: new Date().toLocaleString(),    
}
```